llama_stack/providers/adapters/inference/bedrock/bedrock.py

```diff
@@ -443,16 +445,18 @@ def _create_bedrock_client(config: BedrockConfig) -> BaseClient:
     retries_config = {
         k: v
         for k, v in dict(
-            total_max_attempts=config.total_max_attempts,
-            mode=config.retry_mode,
+            total_max_attempts=os.environ.get(
```

Comment:
we shouldnt rely on env vars in distros. It will be better to take these as provider parameters which are provided at the time of spinning up the distro in run.yaml

Reply:
the main reason for taking in env variables here is that the session token is temporary and if the expiry is short, it requires constantly changing the config to the new value. reading from env can prevent such frequent updates to the config file. In this case, we just prefer the value in the env var. we always fall back to the value specified in the config.

Reply:
@dineshyv But if it is temporary, the issue still persists. While the stack server is running, how is this environment getting updated and provided to the server?!

Reply:
That is true, we still need to restart the server for now to pick it up, but its an improvement over having to change configs every time. The next improvement can be periodically checking if the credentials changed and reinitialize the client.

Reply:
I don't think this is an improvement actually. The central issue is that these credentials are temporary and get revoked. I think this cannot be how things work in a production environment where a server can be running for variable amounts of time. I wonder if there is a way with AWS (and hence in `boto`) where one can periodically re-fresh credentials _while_ the process is running?

Reply:
I dont think the temp credentials is that big of a problem for developers since most of them use IAM roles which have static credentials. This mainly for people logging through SSO which have temp credentials. i am going to revert back to the previous behavior for now. We can think about how to implement this correctly when there is a need.

Reply:
Actually I went back and I thought env var wasn't actually a bad idea :) because you could imagine the server recycler to "re-auth, set vars and restart". And that seems reasonable.

I will let you and @raghotham make the call here.